src/components/Modal/Modal.tsx

```diff
@@ -329,15 +329,16 @@ export const Modal = (props: ModalProps) => {
         // Passing onClose to the Dialog allows it to close the modal when the ESC key is triggered.
         onClose={onClose}
       >
-        <Dialog.Overlay
+        <div
           className={clsx(
             styles['modal__overlay'],
             overlayEmphasis &&
               styles[`modal__overlay--emphasis-${overlayEmphasis}`],
           )}
         />
-
-        <ModalContent onClose={onClose} {...rest} />
+        <Dialog.Panel>
+          <ModalContent onClose={onClose} {...rest} />
+        </Dialog.Panel>
```

Comment:
This is the only typescript error I got from the migration. Dialog.Overlay was deprecated in v1.6

https://github.com/tailwindlabs/headlessui/releases/tag/%40headlessui%2Freact%40v1.6.0